### ISSUE TITLE
How to change severity to warning if CaseCheck fails for rule identifier_name

### ISSUE THREAD

**Issue by vipin04** [realm/SwiftLint#2180]
### New Issue Checklist

- [✔️] Updated SwiftLint to the latest version
- [✔️ I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Environment

* SwiftLint version: 0.25.1
* Installation method used: Cocoapods
* Paste your configuration file:

```yml
disabled_rules:
  - force_cast
#  - function_body_length
#  - trailing_newline
#  - opening_brace
#  - empty_count
#  - comma
#  - colon
#  - type_name
#  - variable_name_min_length
#  - trailing_semicolon
#  - force_try
#  - nesting
#  - variable_name
#  - conditional_binding_cascade
#  - variable_name_max_length
#  - operator_whitespace
#  - control_statement
#  - legacy_constant
#  - line_length
#  - return_arrow_whitespace
  - trailing_whitespace
#  - closing_brace
#  - statement_position
#  - type_body_length
#  - file_length
#  - leading_whitespace
#  - legacy_constructor
  - todo
  # - valid_docs
  # - missing_docs

opt_in_rules: # some rules are only opt-in
#  - empty_count
  # - missing_docs
  - cyclomatic_complexity
  # Find all the available rules by running:
  # swiftlint rules

whitelist_rules:

excluded: # paths to ignore during linting. Takes precedence over `included`.
  - Carthage
  - Pods
  - Vendor

included: # paths to include during linting. `--path` is ignored if present.
  - Source

# configurable rules can be customized from this configuration file
# binary rules can set their severity level
# force_cast: warning # implicitly
force_try:
  severity: warning # explicitly

# rules that have both warning and error levels, can set just the warning level
# implicitly
line_length: 310

# they can set both implicitly with an array
type_body_length:
  - 1400 # warning
  - 2000 # error

function_body_length:
  - 400 # warning
  - 1500 # error

# or they can set both explicitly
file_length:
  warning: 2000
  error: 4000

# naming rules can set warnings/errors for min_length and max_length
# additionally they can set excluded names
type_name:
  min_length: 3 # only warning
  max_length: # warning and error
    warning: 60
    error: 80
  excluded: iPhone # excluded via string

identifier_name:
  severity: warning
  min_length: 1 # only warning
  max_length: 80 # warning 
  excluded: # excluded via string array
    - id
    - URL
  validates_start_with_lowercase: true



function_parameter_count:
  warning: 7
  error: 10

cyclomatic_complexity:
  warning: 60 # 10
  error: 61 # 15

reporter: "xcode" # reporter type (xcode, json, csv, checkstyle)

```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
No
* Which Xcode version are you using (check `xcode-select -p`)?
9.3
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers an error `Identifier Name Violation: Enum element name should start with a lowercase character: 'InvalidURL' (identifier_name)` when all I want is a warning. I do not want to disable the rule:
enum PUNErrorString: String {
    case InvalidURL = "Invalid URL"
    case InvalidRequest = "Invalid URL Request Param"
    case NoInternetConnection = "No Internet Connection"
  }
```

**Comment by marcelofabri** [comment]
This is not possible currently, but PRs are welcome!

**Comment by Sethmr** [comment]
I second wanting this ability. I wanted to put in some temporary code prior to a refactor to test some things and the force_cast being an error was quite irritating. I want the warning to remain so I can correct the code after the refactor, but for now the only option I'm left with is to add force cast to the yml... which is fine, but not optimal.

**Comment by jeetdholakia** [comment]
Are there any updates on this feature?

**Comment by deepesh-vasthimal-cko** [comment]
Any updates on this now ?

**Comment by jpsim** [comment]
Yes this should be possible:

```yaml
identifier_name:
  minLength:
    warning: 2
    error: 1
```

**Comment by rtharston** [comment]
@jpsim the problem in question is setting `severity: warning` doesn't work.

**Comment by jpsim** [comment]
@rtharston please open a new issue with full context and repro steps and I'm happy to look into your issue.

**Comment by mohit-kapadia-at-lirisoft** [comment]
force_cast: warning
or
force_cast:
  severity: warning

Both do not work.

**Comment by conhanley** [comment]
Doesn’t seem to be fixed, `severity: warning` does not work to stop errors from occurring when the lowercase/uppercase violation for `identifer_name` is detected. i.e. it causes my build to fail!

I see this was closed. Noooooooo!

**Comment by conhanley** [comment]
> @rtharston please open a new issue with full context and repro steps and I'm happy to look into your issue.

@jpsim I hope I’ve provided some additional context here beyond what @rtharston and the OP originally mentioned.

**Comment by rtharston** [comment]
Thanks @conhanley for reviving this, and @SimplyDanny for fixing it!
When I first posted I was very new to SwiftLint and the company where I work, so I didn't have much more context, sorry.

But now that this is fixed we can remove `identifier_name` from the disabled_rules list for the first time since we started using SwiftLint three and a half years ago! 😁
